lisa/env.py

```diff
@@ -437,22 +472,31 @@ def get_res_dir(self, name=None, append_time=True, symlink=True):
         """
         logger = self.get_logger()
 
+        if isinstance(self._res_dir, ArtifactPath):
+            root = self._res_dir.root
+            relative = self._res_dir.relative
+        else:
+            root = Path(self._res_dir)
+            relative = Path('')
+
         while True:
             time_str = datetime.now().strftime('%Y%m%d_%H%M%S.%f')
             if not name:
                 name = time_str
             elif append_time:
                 name = "{}-{}".format(name, time_str)
 
-            res_dir = os.path.join(self._res_dir, name)
+            # If we were given an ArtifactPath with an existing root, we
+            # preserve that root so it can be relocated as the caller wants it
+            res_dir = ArtifactPath(root, relative/name)
```

Comment:
As for the use of `ArtifactPath` itself, no direct client of `get_res_dir()` (`TestBundle` & co) have a use for it, but I think that's okay (since it's just a string). You might want to change `TestBundle.from_dir()` though, to make sure we always have the same type for `res_dir`.

Reply:
All direct clients of `get_res_dir()` have a use for it: that's the only reasonable way they can get their res_dir updated when reloading objects from the exekall serialized database. The glue detects these and will relocate them properly.

I could do the same in `TestBundle.from_dir`, although it will unlikely be serialized again (but for consistency that may be better).